src/stream.ts

```diff
@@ -160,11 +155,37 @@ export class WebRTCStream implements Stream {
       if (closed || this.writeClosed) {
         break;
       }
-      let res = buf.subarray();
-      let send_buf = pb.Message.toBinary({ message: buf.subarray() });
-      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${send_buf}`);
-      this.channel.send(send_buf);
+      const res = buf.subarray();
+      const msgbuf = pb.Message.toBinary({message: buf.subarray()});
+      const sendbuf = lp.encode.single(msgbuf)
+      log.trace(`[stream:${this.id}][${this.stat.direction}] sending message: length: ${res.length} ${res}, encoded through pb as ${msgbuf}`);
+      this.channel.send(sendbuf.subarray())
+    }
+  }
+
+  processIncomingProtobuf(buffer: Uint8Array): { data: Uint8Array | undefined } {
+    log.trace(`[stream:${this.id}][${this.stat.direction}] received message: length: ${buffer.length} ${buffer}`);
+    const m = pb.Message.fromBinary(buffer);
+    log.trace(`[stream:${this.id}][${this.stat.direction}] received pb.Message: ${Object.entries(m)}`);
+    switch (m.flag) {
+      case undefined:
+        break; //regular message only
+      case pb.Message_Flag.STOP_SENDING:
+        log.trace('Remote has indicated, with "STOP_SENDING" flag, that it will discard any messages we send.');
+        this.closeWrite();
```

Comment:
I trust you it's fine, but given these are controlled separately without changing the object's structure somehow it does appear fragile to me. Or can you re-open again after closing?